### DISCUSSION TITLE
Why are lifecycle hooks called twice?

### DISCUSSION THREAD

**Discussion by rowild** [quasarframework/quasar#13243] [General - Components / Directives / etc]
I know that there are already some posts concerning this problem, but most of them say that lifecycle hooks get called twice, because "created" can be called on server as well as on client side. 

However, I use Vue 3 with its "setup" method, so there is no "created" hook, and I believe SSR is turned off via quasar.config.js's 

```js
module.exports = configure(function (ctx) {
  return {
    ssr: {    
          pwa: false,
    }
  }
}
```

The structure of my app is as follow:

App.vue pulls in `pages/IndexPage.vue`, which has `layouts/StartpageLayout` as its layout file.

The console shows this.

```shell
[vite] connecting...
client.ts:53 [vite] connected.
client-entry.js?t=1650709506598:51 [Quasar] Running SPA.
App.vue:26 App onMounted invoked

IndexPage.vue:19 IndexPage onBeforeMount invoked
StartpageLayout.vue:54 START_PAGE LAYOUT: onBeforeAppear invoked
StartpageLayout.vue:59 START_PAGE LAYOUT: onAppear invoked
IndexPage.vue:23 IndexPage onMounted invoked
StartpageLayout.vue:28 START_PAGE LAYOUT: onMounted invoked

IndexPage.vue:19 IndexPage onBeforeMount invoked
StartpageLayout.vue:54 START_PAGE LAYOUT: onBeforeAppear invoked
StartpageLayout.vue:59 START_PAGE LAYOUT: onAppear invoked
IndexPage.vue:23 IndexPage onMounted invoked
```

As you can see, the "indexPage" onBeforeMount and almost everthing after that  is called twice. But i do not understand, why?

I prepared a github example, but could not get it working on codesandbox. Therefore I kindly ask those whoare interested in helping me, to please download the repo and star it with `yarn install` and `quasar dev` – Thank you!

https://github.com/rowild/quasar2vue3lifecyclehooksLoadTwiceText

**Comment by pikil** [comment]
That would really help if you can share codesandbox - I could not replicate your issue...
<img width="1260" alt="Screen Shot 2022-04-25 at 9 52 16 PM" src="https://user-images.githubusercontent.com/26779040/165065871-6ef73114-941a-4100-a622-24ab75c19f70.png">

**Reply by rowild** [reply]
Thank you for your reply! It seems you use a webpack setup and not quasar with vite (which I think is the reason, why I cannot get the  example working on codesandbox). Would you be so kind to use my github link and test that one in the meantime?

**Reply by pikil** [reply]
No, I've checked Vite already, webpack just showed the same, so I took the screenshot from there.
Here is the Vite screenshot
<img width="1260" alt="Screen Shot 2022-04-25 at 9 59 26 PM" src="https://user-images.githubusercontent.com/26779040/165067680-ebc8fda7-e191-46cd-983f-f60376d7c60c.png">
